controllers/scaledobject_controller.go

```diff
@@ -224,6 +219,12 @@ func (r *ScaledObjectReconciler) reconcileScaledObject(logger logr.Logger, scale
 
 	// Notify ScaleHandler if a new HPA was created or if ScaledObject was updated
 	if newHPACreated || scaleObjectSpecChanged {
+		// Check for any duplicate names in scaledObject
```

Comment:
+1 Much better place

Reply:
Actually I think, that this might be a better place to implement the uniqueness check, so we don't call `GetScalers(scaledObject)` twice: https://github.com/kedacore/keda/blob/9dfc82a64dc2b6c5d29e9f4e7e70e446efb6e7b6/controllers/hpa.go#L141

Reply:
@zroubalik That is a much much better place! Thanks for the tip! I removed the the uniqueness check function so we need not get all the scalers twice. I updated the tests accordinly.